Improved

Aasyaco opened the pull request.

zrax commented:
I don't know what you're trying to do here, but throwing a bunch of random opcodes into the jump-handling block is probably not very useful...

Aasyaco commented:
> I don't know what you're trying to do here, but throwing a bunch of random opcodes into the jump-handling block is probably not very useful...

Fixed reraise and added some opcodes
